sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -80,7 +80,7 @@ internal static DetectedLanguage ConvertToDetectedLanguage(LanguageDetectionDocu
 
         internal static DetectLanguageResultCollection ConvertToDetectLanguageResultCollection(LanguageDetectionResult results, IDictionary<string, int> idToIndexMap)
         {
-            var detectedLanguages = new List<DetectLanguageResult>(results.Errors.Count);
```

Comment:
I updated this line because I don't think we need to give a length to the list before. When we add the DetectLanguageResults the list will grow to the size needed

Reply:
I also updated this for Recognize Entities and Linked Entities

Reply:
This is true. There is a scenario when, if the list is big, setting the size beforehand is more performant.
See Remarks section [here](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?redirectedfrom=MSDN&view=net-6.0#System_Collections_Generic_List_1__ctor_System_Int32_):
> If the size of the collection can be estimated, using the [List<T>(Int32)](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?view=net-6.0#system-collections-generic-list-1-ctor(system-int32)) constructor and specifying the initial capacity eliminates the need to perform a number of resizing operations while adding elements to the [List<T>](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=net-6.0).)).

I would prefer to leave as is just in case but don't feel strongly about it as the chances for the `Errors` array to be big are small. What do you think?

Reply:
I think if we want to specify the initial length of the list, maybe we can give the results.Documents length instead of the error? I think there will be more documents than errors so this way we can reduce the chance of resizing as well

Reply:
ohhh interesting!!! yes please do that!!!